src/Trakx.CoinGecko.ApiClient/AddCoinGeckoClientExtension.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Net.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Polly;
+using Polly.Contrib.WaitAndRetry;
+using Polly.Extensions.Http;
+using Serilog;
+using Trakx.Utils.DateTimeHelpers;
+
+namespace Trakx.CoinGecko.ApiClient
+{
+    public static partial class AddCoinGeckoClientExtension
+    {
+        public static IServiceCollection AddCoinGeckoClient(
+            this IServiceCollection services, IConfiguration configuration)
+        {
+            services.AddOptions();
+            services.Configure<CoinGeckoApiConfiguration>(
+                configuration.GetSection(nameof(CoinGeckoApiConfiguration)));
+            AddCommonDependencies(services);
+
+            return services;
+        }
+
+        public static IServiceCollection AddCoinGeckoClient(
+            this IServiceCollection services, CoinGeckoApiConfiguration apiConfiguration)
+        {
+            var options = Options.Create(apiConfiguration);
+            services.AddSingleton(options);
+
+            services.AddSingleton<IPingClient, PingClient>();
```

Comment:
Here I am not sure anymore why we don't do this in the library itself anywhere else 🤔
Yet I think it still allows you to simply call `AddXXXClient` and get all dependencies registered! Do you know why this was added?

Reply:
That was my bad understanding of the purpose of the file "ClientRegistrations.tt". Now I get it.
The issue was fixed.